runners/samza/src/main/java/org/apache/beam/runners/samza/translation/SamzaPortablePipelineTranslator.java

```diff
@@ -85,4 +91,23 @@ public static void createConfig(
       }
     }
   }
+
+  public static Set<String> knownUrns() {
+    return TRANSLATORS.keySet();
+  }
+
+  /** Registers Samza translators. */
+  @AutoService(SamzaPortableTranslatorRegistrar.class)
```

Comment:
Seems to me introducing another translator registrar just for portable runner is not needed. The urns are the same, and the logic are already present inside the current translators. Introducing another registrar is more error-prune as the same translator has to be registered twice, in different places. It's also easy to forget registering, as I think we already miss the WindowAssign translation here. I wouldn't recommend having an extra registrar.

Reply:
Synced offline, we agreed to separate the translator registration since registered translators for classic & portable runners diverges significantly.